src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -1,478 +1,424 @@
-      });
-    },
-
-    showNewAppModal: function(event) {
-      if (this.state.modalClass !== null) {
-        return;
-      }
+    }
+  },
 
-      this.setState({
-        modalClass: NewAppModalComponent
-      });
-    },
+  activateTab: function(id) {
+    this.setState({
+      activeTabId: id
+    });
 
-    onTabClick: function(id) {
-      this.setState({
-        activeTabId: id
-      });
+    if (id === tabs[0].id) {
+      this.setPollResource(this.fetchApps);
+    } else if (id === tabs[1].id) {
+      this.setPollResource(this.fetchDeployments);
+    }
+  },
+
+  routeAbout: function () {
+    return (
+      <AboutModalComponent
+        onDestroy={this.modalDestroy}
+        ref="modal" />
+    );
+  },
+
+  routeApps: function(appid, view) {
+    if(!appid || !this.state.activeApp) {
+      return null;
+    }
 
-      if (id === tabs[0].id) {
-        this.setPollResource(this.fetchApps);
-      } else if (id === tabs[1].id) {
-        this.setPollResource(this.fetchDeployments);
-      }
-    },
-
-    render: function() {
-      var modal;
-
-      /* jshint trailing:false, quotmark:false, newcap:false */
-      if (this.state.modalClass === AppModalComponent) {
-        modal = (
-          <AppModalComponent
-            activeTask={this.state.activeTask}
-            appVersionsFetchState={this.state.appVersionsFetchState}
-            destroyApp={this.destroyApp}
-            fetchTasks={this.fetchTasks}
-            fetchAppVersions={this.fetchAppVersions}
-            model={this.state.activeApp}
-            onDestroy={this.handleModalDestroy}
-            onShowTaskDetails={this.handleShowTaskDetails}
-            onShowTaskList={this.handleShowTaskList}
-            onTasksKilled={this.handleTasksKilled}
-            rollBackApp={this.rollbackToAppVersion}
-            scaleApp={this.scaleApp}
-            suspendApp={this.suspendApp}
-            tasksFetchState={this.state.tasksFetchState}
-            ref="modal" />
-        );
-      } else if (this.state.modalClass === NewAppModalComponent) {
-        modal = (
-          <NewAppModalComponent
-            model={this.state.activeApp}
-            onDestroy={this.handleModalDestroy}
-            onCreate={this.handleAppCreate}
-            ref="modal" />
-        );
-      } else if (this.state.modalClass === AboutModalComponent) {
-        modal = (
-          <AboutModalComponent
-            onDestroy={this.handleModalDestroy}
-            ref="modal" />
-        );
-      }
+    return (
+      <AppModalComponent
+        appVersionsFetchState={this.state.appVersionsFetchState}
+        destroyApp={this.destroyApp}
+        fetchAppVersions={this.fetchAppVersions}
+        model={this.state.activeApp}
+        onDestroy={this.modalDestroy}
+        onTasksKilled={this.handleTasksKilled}
+        rollBackApp={this.rollbackToAppVersion}
+        router={this.props.router}
+        ref="modal" />
+    );
+  },
+
+  routeDeployments: function() {
```

Comment:
Can we get rid of this or at least provide the caller with a default option of a `function noop(){}`, so we don't get errors and remove the clutter